onnxruntime/core/providers/cpu/tensor/pad.cc

```diff
@@ -98,10 +117,14 @@ ONNX_CPU_OPERATOR_VERSIONED_KERNEL(
 ONNX_CPU_OPERATOR_KERNEL(
     Pad,
     13,
-    KernelDefBuilder().TypeConstraint(
-        "T",
-        BuildKernelDefConstraintsFromTypeList<Pad11Types>(),
-        BuildKernelDefConstraintsFromTypeList<EnabledPad11Types>()),
+    KernelDefBuilder()
+        .TypeConstraint(
+            "T",
+            BuildKernelDefConstraintsFromTypeList<Pad13Types>(),
+            BuildKernelDefConstraintsFromTypeList<EnabledPad13Types>())
+        .FixedTypeConstraintForHash(
+            "T",
+            BuildKernelDefConstraintsFromTypeList<Pad11Types>()),
     Pad);
 
```

Comment:
This needs to be in 1.8 as it will change the hashes for the opset 13 kernel. We have infrastructure to allow that in the future, but if we get this into 1.8 we don't need to use that yet.

Reply:
Didn't opset-13 ship in previous releases ? If so, shouldn't we keep the hash of this kernel definition backwards compatible ? I followed the recommendation in kernel_def_hash_test.cc to update the types supported by an existing kernel while preserving the hash of the kernel definition or is my understanding incorrect ?

Reply:
Based on offline discussion, fixed in https://github.com/microsoft/onnxruntime/pull/7867